### PR TITLE
Optimize api utils

### PR DESCRIPTION
**Pull request by MariaMoroz**

### CODE REVIEW

File: utils/apiStudyThisUtils.ts [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+        const testId = responseData.id;
+        return { questions: responseData, incorrectOptions, testId };
+    } catch (error) {
+        console.error('Error getting questions:', error);
+        throw error;
+    }
+}
+
+export const answeredQuestions = async (accessToken: string, testId: any, questionId: any, optionId: any) => {
```
Comment (yuliahar) [inline utils/apiStudyThisUtils.ts:62]: Please rename the method (postAnswerToQuestion,  setAnswerToQuestion, etc.)

File: utils/apiStudyThisUtils.ts [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+    try {
+        await makeApiRequest('POST', url, { optionId, questionId }, { authorization: `Bearer ${accessToken}` });
+        console.log('Question answered successfully.');
+    } catch (error) {
+        console.error('Error answering question:', error);
+        throw error;
+    }
+}
+export const finishAnswers = async (accessToken: string, testId: any) => {
```
Comment (yuliahar) [inline utils/apiStudyThisUtils.ts:75]: Fix `any` please

**Review by yuliahar — COMMENTED** [review]
Good job. Please see minor suggestions to fix below.

File: utils/apiCustomerUtils.ts [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+        throw new Error("API_URL environment variable is not set");
+    }
+
+    try {
+        const response = await makeApiRequest('POST', url,
+            {
+                email: `${process.env.ADMIN_EMAIL}`,
+                password: `${process.env.ADMIN_PASSWORD}`,
+            },
+            {});
```
[the remark this answers is not in the document]
Comment (MariaMoroz) [reply utils/apiCustomerUtils.ts:19]: done

File: utils/apiCustomerUtils.ts [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+        expect(response.status).toEqual(201);
+    } catch (error) {
+        throw new Error(`Failed to get admin token: ${error.message}`);
+    }
+}
+
+export const sendCustomerInvite = async () => {
+    const url = '/admin/customer-invites';
+
+    if (!process.env.API_URL) {
```
[the remark this answers is not in the document]
Comment (MariaMoroz) [reply utils/apiCustomerUtils.ts:34]: done

File: utils/apiCustomerUtils.ts [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+        throw new Error("API_URL environment variable is not set");
+    }
+
+    try {
+        let newCustomerEmail = `${process.env.COMMON_EMAIL_PART}${Date.now()}@gmail.com`;
+
+        const response = await makeApiRequest('POST', url,
+            {
+                customerEmail: newCustomerEmail,
+                courseId: "bcf37a9f-af5f-47b0-b9aa-c8e36bbd8278",
```
[the remark this answers is not in the document]
Comment (MariaMoroz) [reply utils/apiCustomerUtils.ts:44]: done

File: utils/apiCustomerUtils.ts [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+        expect(response.status).toEqual(201);
+    } catch (error) {
+        throw new Error(`Failed to send customer invite: ${error.message}`);
+    }
+}
+
+export const confirmCustomerInvite = async () => {
+    const url = "/auth/customer/signUp/promo";
+
+    if (!process.env.API_URL) {
```
[the remark this answers is not in the document]
Comment (MariaMoroz) [reply utils/apiCustomerUtils.ts:64]: done

File: utils/apiCustomerUtils.ts [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+                email: `${process.env.ADMIN_EMAIL}`,
+                password: `${process.env.ADMIN_PASSWORD}`,
+            },
+            {});
+
+        const responseData = await response.json();
+        const accessAdminToken = responseData.token;
+        process.env["ADMIN_TOKEN"] = accessAdminToken;
+
+        expect(response.status).toEqual(201);
```
[the remark this answers is not in the document]
Comment (MariaMoroz) [reply utils/apiCustomerUtils.ts:25]: done

File: utils/apiCustomerUtils.ts [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+            { Authorization: `Bearer ${process.env.ADMIN_TOKEN}` });
+
+        const responseData = await response.json();
+        const newCustomerInviteCode = responseData.code;
+        const newCustomerPassword = responseData.password;
+        process.env["NEW_CUSTOMER_EMAIL"] = newCustomerEmail;
+        process.env["INVITE_CODE"] = newCustomerInviteCode;
+        process.env["NEW_CUSTOMER_PASSWORD"] = newCustomerPassword;
+
+        expect(response.status).toEqual(201);
```
[the remark this answers is not in the document]
Comment (MariaMoroz) [reply utils/apiCustomerUtils.ts:55]: done

File: utils/apiCustomerUtils.ts [hunk 9]
```diff
@@ -0,0 +1,136 @@
...
+        process.env["NEW_CUSTOMER_ID"] = newCustomerId;
+
+        expect(response.status).toEqual(201);
+    } catch (error) {
+        console.error("Error invite confirmation:", error);
+        throw error;
+    }
+}
+
+export const deleteNewCustomer = async () => {
```
[the remark this answers is not in the document]
Comment (MariaMoroz) [reply utils/apiCustomerUtils.ts:88]: не знаю пока как это сделать) оставлю на потом

File: utils/apiStudyThisUtils.ts (hunk 1 above)
Comment (MariaMoroz) [reply utils/apiStudyThisUtils.ts:62]: done

File: utils/apiStudyThisUtils.ts [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+        const responseData = await response.json();
+
+        return responseData;
+    } catch (error) {
+        console.error('Error getting question domains:', error);
+        throw error;
+    }
+}
+
+export const getQuestions = async (accessToken: string, id: any, questionsAmount: number) => {
```
[the remark this answers is not in the document]
Comment (MariaMoroz) [reply utils/apiStudyThisUtils.ts]: done

File: utils/apiStudyThisUtils.ts [hunk 11]
```diff
@@ -0,0 +1,87 @@
...
+    const url = '/tutor-tests/start';
+
+    console.log(`Getting questions with domainIds: ${id}, questionsAmount: ${questionsAmount}`);
+
+    try {
+        const response = await makeApiRequest('POST', url, { domainIds: id, questionsAmount }, { authorization: `Bearer ${accessToken}` });
+
+        const responseData = await response.json();
+
+        const incorrectOptions = responseData.questions.map((question: { id: any; options: any[]; }) => ({
```
[the remark this answers is not in the document]
Comment (MariaMoroz) [reply utils/apiStudyThisUtils.ts]: done

File: utils/apiStudyThisUtils.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (MariaMoroz) [reply utils/apiStudyThisUtils.ts]: done

File: utils/apiStudyThisUtils.ts (hunk 2 above)
Comment (MariaMoroz) [reply utils/apiStudyThisUtils.ts]: done

**Pull request merged**